lib/museum.rb

```diff
@@ -0,0 +1,57 @@
+class Museum
+    attr_reader :name, :exhibits, :patrons
+
+    def initialize(name)
+        @name = name
+        @exhibits = []
+        @patrons = []
+    end
+
+    def add_exhibit(exhibit)
+        @exhibits << exhibit
+    end
+
+    def admit(patron)
+        @patrons << patron
+    end
+
+    def recommend_exhibits(patron) #Selects exhibits based on the patrons interest array.
+        @exhibits.select do |exhibit|
+            patron.interests.include?(exhibit.name)
+        end
+    end
+
+    def patrons_by_exhibit_interest #Creates hash by taking each exhibit, and checking which patrons have interest. Interested patrons hash will have exhibits as keys and patrons as values
+        interested_patrons = {}
+        @exhibits.each do |exhibit|
+            interested_patrons[exhibit] = @patrons.select do |patron|
```

Comment:
GREAT use of select to help you build the hash for patrons_by_exhibit_interest.  Using this enumerable avoids the tricky problem of creating a hash with values that are arrays (it can be difficult to set an array as a default value for a hash).